src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3537,6 +3652,21 @@ export class StudyViewPageStore
         return filter ? filter.values : [];
     }
 
+    public getGenePanelFilters(key: string): string[] {
+        return this.genePanelFilters
+            .filter((f: GenePanelFilter) => {
+                return f.molecularProfileIds
+                    .map(this.extractSuffix)
+                    .includes(extractGenePanelSuffix(key));
+            })
+            .map((f: GenePanelFilter) => f.genePanel);
+    }
+
+    private extractSuffix(molecularProfileId: string): string {
```

Comment:
extractSuffixFromMolecularProfileId.   also there has to be some utility file where we could put this.  it shouldn't be a method as it doesn't reference instance